controllers/etcdcluster_controller.go

```diff
@@ -162,6 +162,41 @@ func (r *EtcdClusterReconciler) getStorageOSClientSecret(ctx context.Context, se
 	return secret, nil
 }
 
+// getDefaultStorage returns the the default storage class in the cluster, if more than one storage
+// class is set to default, the first one discovered is returned. An error is returned if no default
+// storage class is found.
+func (r *EtcdClusterReconciler) getDefaultStorageClass(ctx context.Context) (*kstoragev1.StorageClass, error) {
+	storageClasses := &kstoragev1.StorageClassList{}
+	err := r.List(ctx, storageClasses)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// No storageclass found
+			return nil, nil
+		}
+		// Unexpected error, some other problem?
+		return nil, err
+	}
+
+	for _, storageClass := range storageClasses.Items {
+		if defaultSC, ok := storageClass.GetObjectMeta().GetAnnotations()["storageclass.kubernetes.io/is-default-class"]; ok && defaultSC == "true" {
+			return &storageClass, nil
+		}
+	}
+
+	return nil, fmt.Errorf("no default storage class discovered in cluster")
+}
+
+// getDefaultStorageClassName returns the name of the default storage class in the cluster, if more
+// than one storage class is set to default, the first one discovered is returned. An error is returned
+// if no default storage class is found.
+func (r *EtcdClusterReconciler) getDefaultStorageClassName(ctx context.Context) (string, error) {
```

Comment:
I guess not right now, but I like having the option to get the SC in case we need it in future (seeing as we need to pull it from the API here anyway).